reccmp/isledecomp/formats/elf.py

```diff
@@ -0,0 +1,301 @@
+    EM_TI_ARP32 = 143  # Texas Instruments App. Specific RISC
+    EM_TI_PRU = 144  # Texas Instruments Prog. Realtime Unit
+    EM_MMDSP_PLUS = 160  # STMicroelectronics 64bit VLIW DSP
+    EM_CYPRESS_M8C = 161  # Cypress M8C
+    EM_R32C = 162  # Renesas R32C
+    EM_TRIMEDIA = 163  # NXP Semi. TriMedia
+    EM_QDSP6 = 164  # QUALCOMM DSP6
+    EM_8051 = 165  # Intel 8051 and variants
+    EM_STXP7X = 166  # STMicroelectronics STxP7x
+    EM_NDS32 = 167  # Andes Tech. compact code emb. RISC
+    EM_ECOG1X = 168  # Cyan Technology eCOG1X
+    EM_MAXQ30 = 169  # Dallas Semi. MAXQ30 mc
+    EM_XIMO16 = 170  # New Japan Radio (NJR) 16-bit DSP
+    EM_MANIK = 171  # M2000 Reconfigurable RISC
+    EM_CRAYNV2 = 172  # Cray NV2 vector architecture
+    EM_RX = 173  # Renesas RX
+    EM_METAG = 174  # Imagination Tech. META
+    EM_MCST_ELBRUS = 175  # MCST Elbrus
+    EM_ECOG16 = 176  # Cyan Technology eCOG16
+    EM_CR16 = 177  # National Semi. CompactRISC CR16
+    EM_ETPU = 178  # Freescale Extended Time Processing Unit
+    EM_SLE9X = 179  # Infineon Tech. SLE9X
+    EM_L10M = 180  # Intel L10M
+    EM_K10M = 181  # Intel K10M
+    EM_AARCH64 = 183  # ARM AARCH64
+    EM_AVR32 = 185  # Amtel 32-bit microprocessor
+    EM_STM8 = 186  # STMicroelectronics STM8
+    EM_TILE64 = 187  # Tilera TILE64
+    EM_TILEPRO = 188  # Tilera TILEPro
+    EM_MICROBLAZE = 189  # Xilinx MicroBlaze
+    EM_CUDA = 190  # NVIDIA CUDA
+    EM_TILEGX = 191  # Tilera TILE-Gx
+    EM_CLOUDSHIELD = 192  # CloudShield
+    EM_COREA_1ST = 193  # KIPO-KAIST Core-A 1st gen.
+    EM_COREA_2ND = 194  # KIPO-KAIST Core-A 2nd gen.
+    EM_ARCV2 = 195  # Synopsys ARCv2 ISA.
+    EM_OPEN8 = 196  # Open8 RISC
+    EM_RL78 = 197  # Renesas RL78
+    EM_VIDEOCORE5 = 198  # Broadcom VideoCore V
+    EM_78KOR = 199  # Renesas 78KOR
+    EM_56800EX = 200  # Freescale 56800EX DSC
+    EM_BA1 = 201  # Beyond BA1
+    EM_BA2 = 202  # Beyond BA2
+    EM_XCORE = 203  # XMOS xCORE
+    EM_MCHP_PIC = 204  # Microchip 8-bit PIC(r)
+    EM_INTELGT = 205  # Intel Graphics Technology
+    EM_KM32 = 210  # KM211 KM32
+    EM_KMX32 = 211  # KM211 KMX32
+    EM_EMX16 = 212  # KM211 KMX16
+    EM_EMX8 = 213  # KM211 KMX8
+    EM_KVARC = 214  # KM211 KVARC
+    EM_CDP = 215  # Paneve CDP
+    EM_COGE = 216  # Cognitive Smart Memory Processor
+    EM_COOL = 217  # Bluechip CoolEngine
+    EM_NORC = 218  # Nanoradio Optimized RISC
+    EM_CSR_KALIMBA = 219  # CSR Kalimba
+    EM_Z80 = 220  # Zilog Z80
+    EM_VISIUM = 221  # Controls and Data Services VISIUMcore
+    EM_FT32 = 222  # FTDI Chip FT32
+    EM_MOXIE = 223  # Moxie processor
+    EM_AMDGPU = 224  # AMD GPU
+    EM_RISCV = 243  # RISC-V
+    EM_BPF = 247  # Linux BPF -- in-kernel virtual machine
+    EM_CSKY = 252  # C-SKY
+    EM_LOONGARCH = 258  # LoongArch
+
+
+@dataclass(frozen=True)
+class ElfFileHeader:
+    # Based on elf.h from glibc (https://sourceware.org/git/?p=glibc.git;a=blob;f=elf/elf.h)
```

Comment:
Is this ok docstring?
Or must it be
```suggestion
    """
    Based on elf.h from glibc: https://sourceware.org/git/?p=glibc.git;a=blob;f=elf/elf.h
    """
```

Reply:
Oh, I completely missed that. I suppose the code starting from line 221 is based on `elf.h`, but the code above isn't?

In that case, the comment is good enough in my opinion, but a source for the code above wouldn't hurt. Thanks!

Reply:
I've added a top comment to all file formats.